src/acl/agent.ts

```diff
@@ -218,6 +328,115 @@ function isAgentAclRule(aclRule: unstable_AclRule): boolean {
   return getIriOne(aclRule, acl.agent) !== null;
 }
 
+/**
+ * Create a new ACL Rule with the same ACL values as the input ACL Rule, but having a different IRI.
+ *
+ * Note that non-ACL values will not be copied over.
+ *
+ * @param sourceRule ACL rule to duplicate.
+ */
+function duplicateAclRule(sourceRule: unstable_AclRule): unstable_AclRule {
+  let targetRule = createThing();
+  targetRule = setIri(targetRule, rdf.type, acl.Authorization);
+
+  function copyIris(
+    inputRule: typeof sourceRule,
+    outputRule: typeof targetRule,
+    predicate: IriString
+  ) {
+    return getIriAll(inputRule, predicate).reduce(
+      (outputRule, iriTarget) => addIri(outputRule, predicate, iriTarget),
+      outputRule
+    );
+  }
+
+  targetRule = copyIris(sourceRule, targetRule, acl.accessTo);
+  targetRule = copyIris(sourceRule, targetRule, acl.default);
+  targetRule = copyIris(sourceRule, targetRule, acl.agent);
+  targetRule = copyIris(sourceRule, targetRule, acl.agentGroup);
+  targetRule = copyIris(sourceRule, targetRule, acl.agentClass);
+  targetRule = copyIris(sourceRule, targetRule, acl.origin);
+  targetRule = copyIris(sourceRule, targetRule, acl.mode);
```

Comment:
Given that the ACL predicates are used as a collection in at least one other place (to test for non-ACL quads), should these be grouped in a collection in the constants file ? These lines could then become a single `reduce` on this collection.

Reply:
However, I'd be hesitant to do that: that means the list of constants in `constants.ts` would then serve an additional purpose that is not directly obvious from reading it (i.e. to be an exhaustive list of predicates), possibly leading to unexpected breakage.

(Also, I think the other place you're referring to doesn't actually treat them as a single collection; instead, it further divides them in subgroups depending on their purpose, to verify that one or more of `acl:agent`, `acl:agentGroup` and `acl:agentClass` is defined, _and_ one ore more of `acl:accessTo` and `acl:default`, etc.)

Reply:
Fair enough, it's not like these 7 lines were really an issue :)